pkg/frontend/admin_openshiftcluster_kubernetespods_logs.go

```diff
@@ -0,0 +1,59 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) getAdminKubernetesPodLogs(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+	b, err := f._getAdminKubernetesPodLogs(ctx, r, log)
+
+	if err == nil {
+		w.Header().Set("Content-Type", "text/plain")
+	}
+
+	adminReply(log, w, nil, b, err)
+}
+
+func (f *frontend) _getAdminKubernetesPodLogs(ctx context.Context, r *http.Request, log *logrus.Entry) ([]byte, error) {
+	vars := mux.Vars(r)
+
+	containerName, namespace, name := r.URL.Query().Get("container"), r.URL.Query().Get("namespace"), r.URL.Query().Get("name")
```

Comment:
For the sake of clarity, these variables should be from highest to least importance and `name` needs to be changed as it is too generic: e.g.
`namespace, pod(Name), container(Name) := ....`

Reply:
got it .. changes done ..